backend/src/test/scala/cromwell/backend/standard/callcaching/RootWorkflowHashCacheActorSpec.scala

```diff
@@ -0,0 +1,59 @@
+package cromwell.backend.standard.callcaching
+
+import akka.actor.Props
+import akka.testkit._
+import cromwell.backend.standard.callcaching.RootWorkflowFileHashCacheActor.IoHashCommandWithContext
+import cromwell.core.actor.RobustClientHelper.RequestTimeout
+import cromwell.core.{TestKitSuite, WorkflowId}
+import cromwell.core.callcaching.HashKey
+import cromwell.core.io.DefaultIoCommand.DefaultIoHashCommand
+import cromwell.core.io.IoSuccess
+import cromwell.core.path.DefaultPathBuilder
+import org.scalatest.FlatSpecLike
+
+import scala.concurrent.duration._
+
+class RootWorkflowHashCacheActorSpec extends TestKitSuite("RootWorkflowHashCacheActorSpec") with ImplicitSender
+  with FlatSpecLike {
+
+  private val fakeWorkflowId = WorkflowId.randomId()
+  private val fakeFileName = "fakeFileName"
+
+  it should "properly handle the situation when response from IoActor received after the timeout" in {
+    val ioActorProbe = TestProbe()
+    val rootWorkflowFileHashCacheActor = system.actorOf(Props(new RootWorkflowFileHashCacheActor(ioActorProbe.ref, fakeWorkflowId) {
+      override lazy val defaultIoTimeout = 1.second
+    }))
+
+    val ioHashCommandWithContext = IoHashCommandWithContext(DefaultIoHashCommand(DefaultPathBuilder.build("").get), FileHashContext(HashKey(checkForHitOrMiss = false, List.empty), fakeFileName))
+    rootWorkflowFileHashCacheActor ! ioHashCommandWithContext
+
+    //wait for timeout
+    Thread.sleep(2000)
+
+    EventFilter.info(s"Got reply from IoActor for the file $fakeFileName after the timeout. " +
+      s"Will put it in cache just in case. None of the previous requesters will be notified.", occurrences = 1).intercept {
+      ioActorProbe.send(rootWorkflowFileHashCacheActor, (ioHashCommandWithContext.fileHashContext, IoSuccess(ioHashCommandWithContext.ioHashCommand, "Successful result")))
```

Comment:
seems like this message (or its template) should be a constant in the production code